source/pervasive/seq_lib.rs

```diff
@@ -444,24 +1380,465 @@ pub proof fn seq_to_set_is_finite_broadcast<A>(seq: Seq<A>)
+    ensures
+        !(#[trigger] Seq::<A>::empty().contains(x)),
+{}
+
+// This verified lemma used to be an axiom in the Dafny prelude
+// Note: Dafny only does one way implication, but theoretically it could go both ways
+/// A sequence with length 0 is equivalent to the empty sequence
+pub proof fn lemma_seq_empty_equality<A>(s: Seq<A>)
+    ensures
+        #[trigger] s.len() == 0 ==> s=~= Seq::<A>::empty(),
+{}
+
+// This verified lemma used to be an axiom in the Dafny prelude
+/// The concatenation of two sequences contains only the elements
+/// of the two sequences
+pub proof fn lemma_seq_concat_contains_all_elements<A>(x: Seq<A>, y: Seq<A>, elt: A)
+    ensures
+        #[trigger] (x+y).contains(elt) <==> x.contains(elt) ||  y.contains(elt),
+    decreases
+        x.len(),
+{
+    if x.len() == 0 && y.len() >0 {
+        assert((x+y) =~= y);
+    } else {
+        assert forall |elt: A| #[trigger] x.contains(elt) implies #[trigger] (x+y).contains(elt)
+        by {
+            let index = choose |i: int| 0 <= i < x.len() && x[i] == elt;
+            assert((x+y)[index] == elt);
+        }
+        assert forall |elt: A| #[trigger] y.contains(elt) implies #[trigger] (x+y).contains(elt)
+        by {
+            let index = choose |i: int| 0 <= i < y.len() && y[i] == elt;
+            assert((x+y)[index+x.len()] == elt);
+        }
+    }
+}
+
+// This verified lemma used to be an axiom in the Dafny prelude
+/// After pushing an element onto a sequence, the sequence contains that element
+pub proof fn lemma_seq_contains_after_push<A>(s: Seq<A>, v: A, x: A)
+    ensures 
+        (#[trigger] s.push(v).contains(x) <==> v==x || s.contains(x))
+            && #[trigger] s.push(v).contains(v),
+{
+    assert forall |elt: A| #[trigger] s.contains(elt) implies #[trigger] s.push(v).contains(elt)
+    by {
+        let index = choose |i: int| 0 <= i < s.len() && s[i] == elt;
+        assert(s.push(v)[index] == elt);
+    }
+    assert(s.push(v)[s.len() as int] == v);
+}
+
+// This verified lemma used to be an axiom in the Dafny prelude
+/// The subrange of a sequence contains only the elements within the indices `start` and `stop`
+/// of the original sequence. 
+pub proof fn lemma_seq_subrange_elements<A>(s: Seq<A>, start: int, stop: int, x: A)
+    requires
+        0 <= start <= stop <= s.len(),
+    ensures s.subrange(start,stop).contains(x) <==> 
+        (exists |i: int| 0 <= start <= i < stop <= s.len() && s[i] == x),
+{
+    assert((exists |i: int| 0 <= start <= i < stop <= s.len() && s[i] == x) ==> s.subrange(start,stop).contains(x)) by {
+        if exists |i: int| 0 <= start <= i < stop <= s.len() && s[i] == x
+        {
+            let index = choose |i: int| 0 <= start <= i < stop <= s.len() && s[i] == x;
+            assert(s.subrange(start,stop)[index - start] == s[index]);
+        }
+    }
+}
+
+/************************** Lemmas about Take/Drop ***************************/
+
```

Comment:
Many of these still have extraneous triggers

Reply:
Yes sorry I forgot to take all the triggers off after removing broadcast formal